includes/Handlers/Product/Woo_SOR.php

```diff
@@ -81,12 +81,75 @@ public static function update_catalog_item( CatalogObject $catalog_object, \WC_P
 			$item_data->setReportingCategory( $square_category );
 		}
 
-		$catalog_variations = $item_data->getVariations() ?: array();
+		$attributes = $product->get_attributes();
+
+		$product_variation_ids = $product->get_children();
+		$catalog_variations    = $item_data->getVariations() ? $item_data->getVariations() : array();
 
 		// if dealing with a variable product, try and match the variations
 		if ( $product->is_type( 'variable' ) ) {
 
-			$product_variation_ids = $product->get_children();
+			/**
+			 * If there are multiple variations, it must be a considered as Dynamic Options supported product.
+			 * Create/Update and Assign Dynamic Options only if a product
+			 * has multiple attributes OR options already exists in Square.
+			 */
+			if (
+				count( $attributes ) > 1
+			) {
+				$options_ids            = array();
+				$options_data_transient = get_transient( 'wc_square_options_data' );
+
+				// Set the product as a dynamic options product.
+				update_post_meta( $product->get_id(), '_dynamic_options', true );
+
+				// Loop through the attributes to create options and values at Square.
+				foreach ( $attributes as $attribute_id => $attribute ) {
+
+					$attribute_name = $attribute->get_name(); //wow it gives pa_size which saved me!
```

Comment:
Assuming we don't need this comment:

```suggestion
					$attribute_name = $attribute->get_name();
```